test/VoucherHub.test.ts

```diff
@@ -968,6 +968,119 @@ describe('VoucherHub', function () {
+            // Drain
+            await expect(voucherHubWithAnyoneSigner.drainVoucher(voucherAddress))
+                .to.emit(iexecPocoInstance, 'Transfer')
+                .withArgs(voucherAddress, voucherHubAddress, voucherValue)
+                .to.emit(voucherHub, 'Transfer')
+                .withArgs(voucherAddress, ethers.ZeroAddress, voucherValue)
+                .to.emit(voucherHub, 'VoucherDrained')
+                .withArgs(voucherAddress, voucherValue);
+            expect(await iexecPocoInstance.balanceOf(voucherAddress))
+                .to.equal(await voucherHub.balanceOf(voucherAddress))
+                .to.equal(await voucher.getBalance())
+                .to.equal(0);
+            // Should send RLCs to VoucherHub contract.
+            expect(await iexecPocoInstance.balanceOf(voucherHubAddress)).to.equal(
+                voucherHubRlcBalanceBefore + voucherValue,
+            );
+        });
+
+        it('Should not drain if unknown voucher', async function () {
+            await expect(
+                voucherHubWithVoucherManagerSigner.drainVoucher(
+                    ethers.Wallet.createRandom().address,
+                ),
+            ).to.be.revertedWith('VoucherHub: nothing to drain');
+        });
+
+        it('Should not drain if balance is empty', async function () {
+            // Expire voucher
+            const expirationDate = await voucher.getExpiration();
+            await time.setNextBlockTimestamp(expirationDate + 100n); // after expiration
+            // Drain to empty the voucher from its balance.
+            await voucherHubWithVoucherManagerSigner
+                .drainVoucher(voucherAddress)
+                .then((tx) => tx.wait());
+            // Drain a second time when balance is empty.
+            await expect(
+                voucherHubWithVoucherManagerSigner.drainVoucher(voucherAddress),
+            ).to.be.revertedWith('VoucherHub: nothing to drain');
+        });
+    });
+
+    describe('Withdraw', function () {
+        let receiver: string = ethers.Wallet.createRandom().address;
+
+        beforeEach(async function () {
+            await loadFixture(deployFixture);
+        });
+
+        it('Should withdraw all funds', async function () {
+            expect(await iexecPocoInstance.balanceOf(voucherHubAddress)).to.equal(
+                initVoucherHubBalance,
+            );
+            await expect(
+                voucherHubWithAssetEligibilityManagerSigner.withdraw(
+                    receiver,
+                    initVoucherHubBalance,
+                ),
+            )
+                .to.emit(iexecPocoInstance, 'Transfer')
+                .withArgs(voucherHubAddress, receiver, initVoucherHubBalance);
+            expect(await iexecPocoInstance.balanceOf(voucherHubAddress)).to.equal(0);
+            expect(await iexecPocoInstance.balanceOf(receiver)).to.equal(initVoucherHubBalance);
```

Comment:
Maybe check receiver was zero before the call (but wallet is random so we should already be safe!)

Reply:
Yes we're safe here, receiver is random.